backend/de.metas.handlingunits.base/src/main/java/de/metas/picking/service/impl/HU2PackingItemsAllocator.java

```diff
@@ -491,29 +511,104 @@ private void pickFromVHU(@NonNull final I_M_HU pickFromVHU, @NonNull final Packi
 
 		final I_M_ShipmentSchedule shipmentSchedule = getShipmentScheduleById(packedPart.getShipmentScheduleId());
 
+		final BigDecimal currentQtyToDeliver = shipmentSchedule.getQtyToDeliver();
+		final boolean isOverDelivery = currentQtyToDeliver.compareTo(qtyPacked.toBigDecimal()) < 0;
+
 		// Check over-delivery
-		if (!allowOverDelivery)
+		if (!allowOverDelivery && isOverDelivery && TakeWholeHUEnum.NONE.equals(takeWholeHU.getCode()))
```

Comment:
> TakeWholeHUEnum.NONE.equals(takeWholeHU.getCode())

i think that's NOT OK. U are comparing enum with String.
suggestion: introduce and use TakeWholeHUEnum.isBlaBla() so u will be on the safe side